pyomo/gdp/plugins/bigm.py

```diff
@@ -704,66 +722,144 @@ def _transform_constraint(self, obj, disjunct, bigMargs, arg_list,
+                                                             from_args=True)
+            if not need_lower and not need_upper:
+                return lower, upper
         elif parent in bigMargs:
             m = bigMargs[parent]
-            self.used_args[parent] = m
-            bigm_src[constraint] = (bigMargs, parent)
-            return m
+            (lower, upper, 
+             need_lower, need_upper) = self._process_M_value(m, lower, upper,
+                                                             need_lower,
+                                                             need_upper,
+                                                             bigMargs, parent,
+                                                             constraint_name,
+                                                             from_args=True)
+            if not need_lower and not need_upper:
+                return lower, upper
 
         # use the precomputed traversal up the blocks
         for arg in arg_list:
             for block, val in iteritems(arg):
-                self.used_args[block] = val
-                bigm_src[constraint] = (bigMargs, block)
-                return val
+                (lower, upper, 
+                 need_lower, need_upper) = self._process_M_value(val, lower,
+                                                                 upper,
+                                                                 need_lower,
+                                                                 need_upper,
+                                                                 bigMargs,
+                                                                 block,
+                                                                 constraint_name,
+                                                                 from_args=True)
+                if not need_lower and not need_upper:
+                    return lower, upper
 
         # last check for value for None!
         if None in bigMargs:
             m = bigMargs[None]
-            self.used_args[None] = m
-            bigm_src[constraint] = (bigMargs, None)
-            return m
-        return None
-
-    def _get_M_from_suffixes(self, constraint, suffix_list, bigm_src):
+            (lower, upper, 
+             need_lower, need_upper) = self._process_M_value(m, lower, upper,
+                                                             need_lower,
+                                                             need_upper,
+                                                             bigMargs, None,
+                                                             constraint_name,
+                                                             from_args=True)
+            if not need_lower and not need_upper:
+                return lower, upper
+
+        return lower, upper
+
+    def _get_M_from_suffixes(self, constraint, suffix_list, lower, upper):
```

Comment:
Is this a good name?  When I read above where this is called, it looks like this method completely overrides the M values from the args.  Maybe a better name is `_update_M_from_suffixes`?

Reply:
It doesn't override, but I agree that 'update' is more apt, because it will skip anything already found in args.